Automattic/woocommerce-payments
# fix: Stripe Link autofill on checkout

Fixes https://github.com/Automattic/woocommerce-payments/issues/6881
Fixes https://github.com/Automattic/woocommerce-payments/issues/8019

#### Changes proposed in this Pull Request









Ensuring that Stripe Link correctly autofills blocks & shortcode checkout.
Ensuring that country/state dropdowns and text fields are correctly updated on autofill.
Refactored the implementation to fill both billing and shipping addresses with one function.

#### Testing instructions





Perform the same testing on both blocks and shortcode checkout

- As a merchant, ensure Stripe Link is enabled in the site's WooPayments settings
- As a merchant, ensure "card" is enabled at checkout
- As a customer, ensure you're using "US" as a currency
- As a customer, add items to the cart
- As a customer, go to checkout
- As a customer, ensure you have "Credit Card/Debit Card" as a selected payment method
- As a customer, select a country that uses a non-select state field (e.g.: "Yemen")
- As a customer, click the "Stripe Link" button appearing next to the email field
- Click "autofill"
- Stripe Link billing/shipping values should be autofilled

-------------------


**Post merge**
